src/main/java/org/reactome/addlinks/dataretrieval/ensembl/EnsemblBiomartRetriever.java

```diff
@@ -0,0 +1,232 @@
+package org.reactome.addlinks.dataretrieval.ensembl;
+
+import org.apache.http.HttpException;
+import org.apache.http.HttpStatus;
+import org.json.simple.parser.ParseException;
+import org.reactome.addlinks.EnsemblBiomartUtil;
+import org.reactome.addlinks.dataretrieval.FileRetriever;
+
+import java.io.*;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.time.Duration;
+import java.util.*;
+
+public class EnsemblBiomartRetriever extends FileRetriever {
+
+    private Properties properties;
+
+    public EnsemblBiomartRetriever() { }
+
+    public EnsemblBiomartRetriever(String retrieverName)
+    {
+        super(retrieverName);
+    }
+
+    private static final String  microarrayTypesBaseQuery = "type=listAttributes&mart=ENSEMBL_MART_ENSEMBL&virtualSchema=default&dataset=BIOMART_SPECIES_NAME_gene_ensembl&interface=default&attributePage=feature_page&attributeGroup=external&attributeCollection=microarray";
```

Comment:
For readability of the query parameters, maybe this can be broken up into smaller concatenated Strings after each "&".  Something like this:

```
private static final String microarrayTypesBaseQuery = String.join("&",
    "type=listAttributes",
    "mart=ENSEMBL_MART_ENSEMBL",
    "virtualSchema=default",
    "dataset=BIOMART_SPECIES_NAME_gene_ensembl",
    "interface=default",
    "attributePage=feature_page",
    "attributeGroup=external",
    "attributeCollection=microarray"
);
```

Reply:
Yep thats much nicer, made that change